internalsdk/auth/auth.go

```diff
@@ -0,0 +1,214 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/lantern-client/internalsdk/common"
+	"github.com/getlantern/lantern-client/internalsdk/protos"
+	"github.com/getlantern/lantern-client/internalsdk/webclient"
+	"github.com/getlantern/lantern-client/internalsdk/webclient/defaultwebclient"
+	"github.com/go-resty/resty/v2"
+)
+
+var (
+	log = golog.LoggerFor("authclient")
+)
+
+type authClient struct {
+	webclient webclient.RESTClient
+}
+
+type AuthClient interface {
+	//Sign up methods
+	SignUp(ctx context.Context, signupData *protos.SignupRequest) (bool, error)
+	SignupEmailResendCode(ctx context.Context, data *protos.SignupEmailResendRequest) (bool, error)
+	SignupEmailConfirmation(ctx context.Context, data *protos.ConfirmSignupRequest) (bool, error)
+
+	//Login methods
+	GetSalt(ctx context.Context, email string) (*protos.GetSaltResponse, error)
+	LoginPrepare(ctx context.Context, loginData *protos.PrepareRequest) (*protos.PrepareResponse, error)
+	Login(ctx context.Context, loginData *protos.LoginRequest) (*protos.LoginResponse, error)
+	// Recovery methods
+	StartRecoveryByEmail(ctx context.Context, loginData *protos.StartRecoveryByEmailRequest) (bool, error)
+	CompleteRecoveryByEmail(ctx context.Context, loginData *protos.CompleteRecoveryByEmailRequest) (bool, error)
+	ValidateEmailRecoveryCode(ctx context.Context, loginData *protos.ValidateRecoveryCodeRequest) (*protos.ValidateRecoveryCodeResponse, error)
+	// Change email methods
+	ChangeEmail(ctx context.Context, loginData *protos.ChangeEmailRequest) (bool, error)
+	// Complete change email methods
+	CompleteChangeEmail(ctx context.Context, loginData *protos.CompleteChangeEmailRequest) (bool, error)
+	DeleteAccount(ctc context.Context, loginData *protos.DeleteUserRequest) (bool, error)
+
+	//Logout
+	SignOut(ctx context.Context, logoutData *protos.LogoutRequest) (bool, error)
+}
+
+// NewClient creates a new instance of AuthClient
+func NewClient(baseURL string, opts *webclient.Opts) AuthClient {
+	httpClient := opts.HttpClient
+	if httpClient == nil {
+		httpClient = &http.Client{}
+	}
+	webclient := webclient.NewRESTClient(defaultwebclient.SendToURL(httpClient, baseURL, func(client *resty.Client, req *resty.Request) error {
+		req.SetHeader(common.ContentType, "application/x-protobuf")
+		if req.RawRequest.URL != nil && strings.HasPrefix(req.RawRequest.URL.Path, "/users/salt") {
```

Comment:
I noticed we weren't setting any headers when calling the `/users/salt` endpoint. If that was unintentional, you can just remove these lines